### ISSUE TITLE
Breaks backward compatibility with a minor version number change

### ISSUE THREAD

**Issue by marcelalburg** [zalando-stups/stups-spring-oauth2-support#44]
Hello,

#43 breaks the Backwards compatibility. After we changed from 1.0.10 to 1.0.15 we get problems with this code. 

```
OLD:
AbstractAuthenticationExtractor.class
public OAuth2Authentication extractAuthentication(final Map<String, Object> map, final String clientId) {

NEW:
AbstractAuthenticationExtractor.class
 public OAuth2Authentication extractAuthentication(final Map<String, Object> tokenInfoMap, final String clientId,
            final UserRolesProvider userRolesProvider) {

```

We'll change it in our code, but it's hard to fullfill your request to change to the lastest version and inside a minor version change, the structure is broken after it.

**Comment by jbellmann** [comment]
Thanks for reporting the issue. I will have a look.

**Comment by jbellmann** [comment]
Did you found more breaking changes? Other than the code above?

**Comment by marcelalburg** [comment]
Give us some more time :) I'll come back in max 2. days.

**Comment by jbellmann** [comment]
Created an PR to keep backward compatibility. If you will not find any more code affected the following changes should make it compatible again?

[AbstractAuthenticationExtractor](https://github.com/zalando-stups/stups-spring-oauth2-support/pull/45/files#diff-1a891adff07e68cfc37cdbe3323f1771R49)

[AuthenticationExtractor](https://github.com/zalando-stups/stups-spring-oauth2-support/pull/45/files#diff-54b4c8dcc7bbf68728061093a6189aa6R31)

should work then, right?

Regards
Jörg

**Comment by marcelalburg** [comment]
We changed our logic in this class (see above) and it works for us now. From my POV it's the only place you need to change.

Thanks
M

**Comment by jbellmann** [comment]
@marcelalburg 

please try '1.0.15-GH-44'. Maybe it's backward compatible and we can avoid bad experience for other teams who want to upgrade.

Regards
Jörg

**Comment by marcelalburg** [comment]
We already changed our code. I cannot commit into your repo. Please can you create this test into your branch ?

```java
package org.zalando.stups.oauth2.spring.server;

import static org.junit.Assert.*;

import java.lang.reflect.Method;
import java.util.Map;

import org.junit.Test;

public class BackwardCompatibilityTest {

	/*
	 * This test checks the backwards compatibility for the
	 * org.zalando.stups.oauth2.spring.server.DefaultAuthenticationExtractor
	 * 
	 * this test is for version 1.0.x
	 */
	@Test
	public void checkingAbstractAuthenticationExtractor() throws NoSuchMethodException, SecurityException {
		Method methodSuperClazz = null;
		Class cls = DefaultAuthenticationExtractor.class;
		do {
			try {
				methodSuperClazz = cls.getMethod("extractAuthentication", Map.class, String.class);
			} catch (NoSuchMethodException | SecurityException e) {

			}
			cls = cls.getSuperclass();
		} while (cls != null && methodSuperClazz == null);
		
		assertNotNull(methodSuperClazz);
	}
}
```

**Comment by jbellmann** [comment]
I released 1.0.18. Please try out and let me know if it's not working for you.

Regards
Jörg
